controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -867,9 +866,11 @@ public void setSuspension(DeploymentId deploymentId, boolean suspend) {
     /** Deactivate application in the given zone. Even if the application itself does not exist, deactivation of the deployment will still be attempted */
     public void deactivate(ApplicationId instanceId, ZoneId zone) {
         TenantAndApplicationId applicationId = TenantAndApplicationId.from(instanceId);
-        try (Mutex lock = lock(applicationId)) {
-            Optional<LockedApplication> application = getApplication(applicationId).map(app -> new LockedApplication(app, lock));
-            deactivate(instanceId, zone, application).ifPresent(this::store);
+        try (Mutex deploymentLock = lockForDeployment(instanceId, zone)) {
```

Comment:
Not related, but a tiny bug that should be fixed anyway. I doubt it often matters.